staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go

```diff
@@ -232,3 +228,51 @@ func RecordStorageListMetrics(resource string, numFetched, numEvald, numReturned
 	listStorageNumSelectorEvals.WithLabelValues(resource).Add(float64(numEvald))
 	listStorageNumReturned.WithLabelValues(resource).Add(float64(numReturned))
 }
+
+type Monitor interface {
+	Monitor(ctx context.Context) (StorageMetrics, error)
+	Close() error
+}
+
+type StorageMetrics struct {
+	Size int64
+}
+
+type monitorCollector struct {
+	compbasemetrics.BaseStableCollector
+
+	monitorGetter func() ([]Monitor, error)
+}
+
+// DescribeWithStability implements compbasemetrics.StableColletor
+func (c *monitorCollector) DescribeWithStability(ch chan<- *compbasemetrics.Desc) {
+	ch <- storageSizeDescription
+}
+
+// CollectWithStability implements compbasemetrics.StableColletor
+func (c *monitorCollector) CollectWithStability(ch chan<- compbasemetrics.Metric) {
+	monitors, err := c.monitorGetter()
+	if err != nil {
+		return
+	}
+
+	for i, m := range monitors {
+		server := fmt.Sprintf("etcd-%d", i)
```

Comment:
I don't think we will be able to correlate this label with any other one from the other metrics since AFAIK this is the only place we have something like that. So wouldn't having an endpoint like before be easier to use?

Reply:
No, it's third place to do that. It's also done by apiserver health probes and apiserver component status. As for usefulness of endpoint, it's worse as apiserver will expose all etcd members it connects to. For etcd HA clusters thats  a 3 endpoints with the same value (+/- time shift). For each API override that's additional 3 endpoints. 

The semantic meaning of server is:
* `etcd-0` the default etcd
* `etcd-1` the first API Resource override, based on order of flags provided to apiserver.
* `etcd-2` the second API Resource override, and so forth.

Reply:
Makes sense, thank you for the clarification :+1: